lib/xmldsig/signature.rb

```diff
@@ -2,10 +2,11 @@ module Xmldsig
   class Signature
     attr_accessor :signature
 
-    def initialize(signature, id_attr = nil, referenced_documents = {})
+    def initialize(signature, id_attr = nil, referenced_documents = {}, base64_strict_encode = false)
```

Comment:
Is this an internal class? Surprised to see this new option leak out in more than one place.

Reply:
The entry point for using `xmldsig` is the `Xmldsig::SignedDocument` class. It delegates signing requests to two internal classes `Xmldsig::Signature` (SignatureValue) and `Xmldsig::Reference` (DigestValue), hence the additional option passing. In practical terms it makes sense that `Signature` and `Reference` would be encapsulated and not directly accessible in the API.

Reply:
Yep, make sense. If `Signature` and `Reference` are internal classes, though, can their signature not have a default `base64_strict_encode` value then? This keeps things explicit for internal use while still preserving the desired external interaction.

Reply:
I agree that this would be a better approach. There is a challenge in doing so as the `sign` method for both `Signature` and `Reference` delegate further to `(signature | digest)_value` setter method (see https://github.com/spreedly/xmldsig/pull/1/files#diff-76dcac0892623c31e18d91e142bf07ccL100). I'd have to do a slight refactor to make it work and balancing time-value trade-offs, I'd argue against spending the time to do it.

Having the option be part of the object initialization follows the existing option pattern used by the gem.

Reply:
👍